packages/vitest/src/node/core.ts

```diff
@@ -592,6 +592,8 @@ export class Vitest {
       id = slash(id)
       updateLastChanged(id)
       if (await this.isTargetFile(id)) {
+        this.projectsTestFiles.set(id, new Set(this.projects))
```

Comment:
This is not true most of the time when using workspaces. `this.projectsTestFiles` holds every test file and its respective projects. With these two projects it will break:

```ts
[
  // only first two should be in a set
  { test: { include: ['test1/**/*'], environment: 'jsdom' } },
  { test: { include: ['test1/**/*'], environment: 'node' } },
  { test: { include: ['test2/**/*'] } },
]
```